offline_converter.py

```diff
@@ -253,15 +254,22 @@ def _convert(self, project: QgsProject) -> None:
 
             if layer_action == SyncAction.OFFLINE:
                 if self.project_configuration.offline_copy_only_aoi:
-                    extent = QgsCoordinateTransform(
-                        QgsCoordinateReferenceSystem(self.area_of_interest_crs),
-                        layer.crs(),
-                        QgsProject.instance(),
-                    ).transformBoundingBox(self.area_of_interest.boundingBox())
-                    layer.selectByRect(extent)
-
-                if not layer.selectedFeatureCount():
-                    layer.selectByIds([FID_NULL])
+                    if (
+                        layer.geometryType() is not Qgis.GeometryType.Null
+                        and layer.geometryType() is not Qgis.GeometryType.Unknown
+                    ):
+                        extent = QgsCoordinateTransform(
+                            QgsCoordinateReferenceSystem(self.area_of_interest_crs),
+                            layer.crs(),
+                            QgsProject.instance(),
+                        ).transformBoundingBox(self.area_of_interest.boundingBox())
+                        layer.selectByRect(extent)
+
+                        if not layer.selectedFeatureCount():
+                            layer.selectByIds([FID_NULL])
+                    else:
+                        # insure that geometry-less layers do not have selected features that would interfere with the process
+                        layer.removeSelection()
```

Comment:
I would have expected a select all, does this work?

Reply:
@m-kuhn , it works, the logic is in QgsOfflineEditing, and it only extracts selected features when 1 or more is selected, see: https://github.com/qgis/QGIS/blob/master/src/core/qgsofflineediting.cpp#L788C5-L793